sources/api/storewolf/src/main.rs

```diff
@@ -186,10 +185,12 @@ fn parse_metadata_toml(md_toml_val: toml::Value) -> Result<Vec<model::Metadata>>
                 );
 
                 // Ensure the metadata/data keys don't contain newline chars
-                let md =
-                    SingleLineString::try_from(md_key).context(error::SingleLineStringSnafu)?;
-                let key =
-                    SingleLineString::try_from(data_key).context(error::SingleLineStringSnafu)?;
+                let md = md_key
+                    .parse::<SingleLineString>()
+                    .context(error::SingleLineStringSnafu)?;
+                let key = data_key
+                    .parse::<SingleLineString>()
+                    .context(error::SingleLineStringSnafu)?;
```

Comment:
It looks like we should be able to unit test this function to make sure this hasn't changed behavior

Reply:
If you mean `.parse::<SingleLineString>()`, I think the unit test is located here: https://github.com/bottlerocket-os/bottlerocket/blob/cf7a49a705883bf5377f3a6de746aa8642a0a2ca/sources/models/src/modeled_types/shared.rs#L101-L129

Unless you meant something else?

Reply:
I mean `fn parse_metadata_toml`. The tests above make sure that `parse` works, but, is it exactly equivalent to the previously used `try_from` call?

Reply:
We agreed to open an issue for then entire main.rs file since it's missing tests completely.